property/index.js

```diff
@@ -78,6 +80,7 @@ module.exports = yeoman.generators.Base.extend({
         name: 'type',
         message: 'Property type:',
         type: 'list',
+        default: this.propDefinition && this.propDefinition.type,
```

Comment:
When the type entered in the last run was an array, then `propDefinition.type` will be array-of-something, e.g. `['string']` and thus it needs to be converted back to "array" string here.

I was not able to verify this, because when I try to enter an invalid value for array of numbers, the property is still created - see https://github.com/strongloop/generator-loopback/pull/200#issuecomment-228074338.